code/_helpers/protection.dm

```diff
@@ -0,0 +1,16 @@
+/atom/proc/isProtected(var/mob/user)
+	var/area/protected/A
+	if(istype(src, /turf))
+		A = get_area(src)
+	else
+		A = get_area(loc)
+		
+	if(!A.safe_zone)
+		return FALSE
+	if(!user)
+		return TRUE
+	// Check permission whitelists for this user.
+	return TRUE // TODO: actually implement.
+
+/area/protected
```

Comment:
You don't typecheck `A`. This should really be an area flag rather than a bool.

Reply:
I wasn't aware of area_flags, I'll see about converting it, actually.

Reply:
Converted.